### PR TITLE
[MXNET-1287] Feat dep

### PR DESCRIPTION
**Pull request by zachgk**
## Description ##
This PR fixes the feature/deprecation warnings and makes them display as part of the Maven run by default. Most of the fixes are related to the change to IndexedSeq[DataDesc]. Contains a part of #13995.

@lanking520 @andrewfayres 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by zachgk** [comment]
@mxnet-label-bot add [Scala, Maven, pr-awaiting-review]

File: scala-package/core/src/main/scala/org/apache/mxnet/io/PrefetchingIter.scala [hunk 1]
```diff
@@ -42,71 +42,51 @@ class PrefetchingIter(
...
-      }
+      iters.map(_.provideData).zip(dataNames).map { case (providedData, names) =>
+        providedData.map { case (oldName, shape) => names(oldName) -> shape }
+      }.reduceLeft(_ ++ _)
     }
   }
 
-  private val _provideLabel: ListMap[String, Shape] = {
+  @deprecated("Please use provideDataDesc instead", "1.3.0")
+  override def provideLabel: ListMap[String, Shape] = {
```
Comment (lanking520) [inline scala-package/core/src/main/scala/org/apache/mxnet/io/PrefetchingIter.scala]: Why not make it private?

File: scala-package/examples/src/main/scala/org/apache/mxnetexamples/neuralstyle/NeuralStyle.scala [hunk 2]
```diff
@@ -39,7 +39,7 @@ object NeuralStyle {
   private val logger = LoggerFactory.getLogger(classOf[NeuralStyle])
 
   def preprocessContentImage(path: String, longEdge: Int, ctx: Context): NDArray = {
-    val img = Image(new File(path))
+    val img = Image.fromFile(new File(path))
```
Comment (lanking520) [inline scala-package/examples/src/main/scala/org/apache/mxnetexamples/neuralstyle/NeuralStyle.scala]: Why change this?

File: scala-package/core/src/main/scala/org/apache/mxnet/io/PrefetchingIter.scala (hunk 1 above)
Comment (zachgk) [reply scala-package/core/src/main/scala/org/apache/mxnet/io/PrefetchingIter.scala]: The _provideLabel was a private implementation of the externally accessible provideLabel. I thought there was no point in having both so I combined them and left the public one.

File: scala-package/examples/src/main/scala/org/apache/mxnetexamples/neuralstyle/NeuralStyle.scala (hunk 2 above)
Comment (zachgk) [reply scala-package/examples/src/main/scala/org/apache/mxnetexamples/neuralstyle/NeuralStyle.scala]: There was a warning against using Image(). I don't remember what the warning was, though.

**Comment by lanking520** [comment]
Does the Scala GPU failure related to your changes?

**Pull request merged**